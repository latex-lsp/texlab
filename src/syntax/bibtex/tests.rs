use insta::assert_debug_snapshot;

use super::{parse, SyntaxNode};

fn setup(input: &str) -> SyntaxNode {
    SyntaxNode::new_root(parse(input))
}

#[test]
fn test_preamble() {
    assert_debug_snapshot!(setup(r#"@preamble{"Hello World!"}"#));
}

#[test]
fn test_comment() {
    assert_debug_snapshot!(setup(
        r#"@comment{aksin,
    author       = {Aks{\i}n, {\"O}zge and T{\"u}rkmen, Hayati and Artok, Levent
                    and {\c{C}}etinkaya, Bekir and Ni, Chaoying and
                    B{\"u}y{\"u}kg{\"u}ng{\"o}r, Orhan and {\"O}zkal, Erhan},
    title        = {Effect of immobilization on catalytic characteristics of
                    saturated {Pd-N}-heterocyclic carbenes in {Mizoroki-Heck}
                    reactions},
    journaltitle = jomch,
    date         = 2006,
    volume       = 691,
    number       = 13,
    pages        = {3027-3036},
    indextitle   = {Effect of immobilization on catalytic characteristics},
}"#
    ));
}

#[test]
fn test_entry_colon() {
    assert_debug_snapshot!(setup(
        r#"@article{foo:2019,
    author = {Foo Bar},
    title = {Baz Qux},
    year = {2019},
}"#
    ));
}

#[test]
fn test_biblatex_examples_001() {
    assert_debug_snapshot!(setup(
        r#"@incollection{westfahl:space,
    author       = {Westfahl, Gary},
    title        = {The True Frontier},
    subtitle     = {Confronting and Avoiding the Realities of Space in {American}
                    Science Fiction Films},
    pages        = {55-65},
    crossref     = {westfahl:frontier},
    langid       = {english},
    langidopts   = {variant=american},
    indextitle   = {True Frontier, The},
    annotation   = {A cross-referenced article from a \texttt{collection}. This is
                    an \texttt{incollection} entry with a \texttt{crossref}
                    field. Note the \texttt{subtitle} and \texttt{indextitle}
                    fields},
}

% booktitle and booksubtitle are only needed for BibTeX's less sophisticated
% inheritance set-up to make sure westfahl:space shows correctly.
% With Biber they are not needed.
@collection{westfahl:frontier,
    editor       = {Westfahl, Gary},
    title        = {Space and Beyond},
    date         = 2000,
    subtitle     = {The Frontier Theme in Science Fiction},
    publisher    = {Greenwood},
    location     = {Westport, Conn. and London},
    langid       = {english},
    langidopts   = {variant=american},
    booktitle    = {Space and Beyond},
    booksubtitle = {The Frontier Theme in Science Fiction},
    annotation   = {This is a \texttt{collection} entry. Note the format of the
                    \texttt{location} field as well as the \texttt{subtitle}
                    field},
}"#
    ));
}

#[test]
fn test_biblatex_examples_002() {
    assert_debug_snapshot!(setup(
        r#"@string{jomch   = {J.~Organomet. Chem.}}

@article{aksin,
    author       = {Aks{\i}n, {\"O}zge and T{\"u}rkmen, Hayati and Artok, Levent
                    and {\c{C}}etinkaya, Bekir and Ni, Chaoying and
                    B{\"u}y{\"u}kg{\"u}ng{\"o}r, Orhan and {\"O}zkal, Erhan},
    title        = {Effect of immobilization on catalytic characteristics of
                    saturated {Pd-N}-heterocyclic carbenes in {Mizoroki-Heck}
                    reactions},
    journaltitle = jomch,
    date         = 2006,
    volume       = 691,
    number       = 13,
    pages        = {3027-3036},
    indextitle   = {Effect of immobilization on catalytic characteristics},
}"#
    ));
}

#[test]
fn test_biblatex_examples_003() {
    assert_debug_snapshot!(setup(
        r#"@article{angenendt,
    author       = {Angenendt, Arnold},
    title        = {In Honore Salvatoris~-- Vom Sinn und Unsinn der
                    Patrozinienkunde},
    journaltitle = {Revue d'Histoire Eccl{\'e}siastique},
    date         = 2002,
    volume       = 97,
    pages        = {431--456, 791--823},
    langid       = {german},
    indextitle   = {In Honore Salvatoris},
    shorttitle   = {In Honore Salvatoris},
    annotation   = {A German article in a French journal. Apart from that, a
                    typical \texttt{article} entry. Note the \texttt{indextitle}
                    field},
}"#
    ));
}
