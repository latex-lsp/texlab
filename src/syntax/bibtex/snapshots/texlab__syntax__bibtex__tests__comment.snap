---
source: src/syntax/bibtex/tests.rs
assertion_line: 16
expression: "setup(r#\"@comment{aksin,\n    author       = {Aks{\\i}n, {\\\"O}zge and T{\\\"u}rkmen, Hayati and Artok, Levent\n                    and {\\c{C}}etinkaya, Bekir and Ni, Chaoying and\n                    B{\\\"u}y{\\\"u}kg{\\\"u}ng{\\\"o}r, Orhan and {\\\"O}zkal, Erhan},\n    title        = {Effect of immobilization on catalytic characteristics of\n                    saturated {Pd-N}-heterocyclic carbenes in {Mizoroki-Heck}\n                    reactions},\n    journaltitle = jomch,\n    date         = 2006,\n    volume       = 691,\n    number       = 13,\n    pages        = {3027-3036},\n    indextitle   = {Effect of immobilization on catalytic characteristics},\n}\"#)"
---
ROOT@0..637
  COMMENT@0..637
    TYPE@0..8 "@comment"
    L_CURLY@8..9 "{"
    WORD@9..14 "aksin"
    COMMA@14..15 ","
    WHITESPACE@15..20 "\n    "
    WORD@20..26 "author"
    WHITESPACE@26..33 "       "
    EQ@33..34 "="
    WHITESPACE@34..35 " "
    L_CURLY@35..36 "{"
    WORD@36..39 "Aks"
    L_CURLY@39..40 "{"
    COMMAND_NAME@40..42 "\\i"
    R_CURLY@42..43 "}"
    WORD@43..44 "n"
    COMMA@44..45 ","
    WHITESPACE@45..46 " "
    L_CURLY@46..47 "{"
    COMMAND_NAME@47..49 "\\\""
    WORD@49..50 "O"
    R_CURLY@50..51 "}"
    WORD@51..54 "zge"
    WHITESPACE@54..55 " "
    WORD@55..58 "and"
    WHITESPACE@58..59 " "
    WORD@59..60 "T"
    L_CURLY@60..61 "{"
    COMMAND_NAME@61..63 "\\\""
    WORD@63..64 "u"
    R_CURLY@64..65 "}"
    WORD@65..70 "rkmen"
    COMMA@70..71 ","
    WHITESPACE@71..72 " "
    WORD@72..78 "Hayati"
    WHITESPACE@78..79 " "
    WORD@79..82 "and"
    WHITESPACE@82..83 " "
    WORD@83..88 "Artok"
    COMMA@88..89 ","
    WHITESPACE@89..90 " "
    WORD@90..96 "Levent"
    WHITESPACE@96..117 "\n                    "
    WORD@117..120 "and"
    WHITESPACE@120..121 " "
    L_CURLY@121..122 "{"
    COMMAND_NAME@122..124 "\\c"
    L_CURLY@124..125 "{"
    WORD@125..126 "C"
    R_CURLY@126..127 "}"
    R_CURLY@127..128 "}"
    WORD@128..136 "etinkaya"
    COMMA@136..137 ","
    WHITESPACE@137..138 " "
    WORD@138..143 "Bekir"
    WHITESPACE@143..144 " "
    WORD@144..147 "and"
    WHITESPACE@147..148 " "
    WORD@148..150 "Ni"
    COMMA@150..151 ","
    WHITESPACE@151..152 " "
    WORD@152..160 "Chaoying"
    WHITESPACE@160..161 " "
    WORD@161..164 "and"
    WHITESPACE@164..185 "\n                    "
    WORD@185..186 "B"
    L_CURLY@186..187 "{"
    COMMAND_NAME@187..189 "\\\""
    WORD@189..190 "u"
    R_CURLY@190..191 "}"
    WORD@191..192 "y"
    L_CURLY@192..193 "{"
    COMMAND_NAME@193..195 "\\\""
    WORD@195..196 "u"
    R_CURLY@196..197 "}"
    WORD@197..199 "kg"
    L_CURLY@199..200 "{"
    COMMAND_NAME@200..202 "\\\""
    WORD@202..203 "u"
    R_CURLY@203..204 "}"
    WORD@204..206 "ng"
    L_CURLY@206..207 "{"
    COMMAND_NAME@207..209 "\\\""
    WORD@209..210 "o"
    R_CURLY@210..211 "}"
    WORD@211..212 "r"
    COMMA@212..213 ","
    WHITESPACE@213..214 " "
    WORD@214..219 "Orhan"
    WHITESPACE@219..220 " "
    WORD@220..223 "and"
    WHITESPACE@223..224 " "
    L_CURLY@224..225 "{"
    COMMAND_NAME@225..227 "\\\""
    WORD@227..228 "O"
    R_CURLY@228..229 "}"
    WORD@229..233 "zkal"
    COMMA@233..234 ","
    WHITESPACE@234..235 " "
    WORD@235..240 "Erhan"
    R_CURLY@240..241 "}"
    COMMA@241..242 ","
    WHITESPACE@242..247 "\n    "
    WORD@247..252 "title"
    WHITESPACE@252..260 "        "
    EQ@260..261 "="
    WHITESPACE@261..262 " "
    L_CURLY@262..263 "{"
    WORD@263..269 "Effect"
    WHITESPACE@269..270 " "
    WORD@270..272 "of"
    WHITESPACE@272..273 " "
    WORD@273..287 "immobilization"
    WHITESPACE@287..288 " "
    WORD@288..290 "on"
    WHITESPACE@290..291 " "
    WORD@291..300 "catalytic"
    WHITESPACE@300..301 " "
    WORD@301..316 "characteristics"
    WHITESPACE@316..317 " "
    WORD@317..319 "of"
    WHITESPACE@319..340 "\n                    "
    WORD@340..349 "saturated"
    WHITESPACE@349..350 " "
    L_CURLY@350..351 "{"
    WORD@351..355 "Pd-N"
    R_CURLY@355..356 "}"
    WORD@356..369 "-heterocyclic"
    WHITESPACE@369..370 " "
    WORD@370..378 "carbenes"
    WHITESPACE@378..379 " "
    WORD@379..381 "in"
    WHITESPACE@381..382 " "
    L_CURLY@382..383 "{"
    WORD@383..396 "Mizoroki-Heck"
    R_CURLY@396..397 "}"
    WHITESPACE@397..418 "\n                    "
    WORD@418..427 "reactions"
    R_CURLY@427..428 "}"
    COMMA@428..429 ","
    WHITESPACE@429..434 "\n    "
    WORD@434..446 "journaltitle"
    WHITESPACE@446..447 " "
    EQ@447..448 "="
    WHITESPACE@448..449 " "
    WORD@449..454 "jomch"
    COMMA@454..455 ","
    WHITESPACE@455..460 "\n    "
    WORD@460..464 "date"
    WHITESPACE@464..473 "         "
    EQ@473..474 "="
    WHITESPACE@474..475 " "
    INTEGER@475..479 "2006"
    COMMA@479..480 ","
    WHITESPACE@480..485 "\n    "
    WORD@485..491 "volume"
    WHITESPACE@491..498 "       "
    EQ@498..499 "="
    WHITESPACE@499..500 " "
    INTEGER@500..503 "691"
    COMMA@503..504 ","
    WHITESPACE@504..509 "\n    "
    WORD@509..515 "number"
    WHITESPACE@515..522 "       "
    EQ@522..523 "="
    WHITESPACE@523..524 " "
    INTEGER@524..526 "13"
    COMMA@526..527 ","
    WHITESPACE@527..532 "\n    "
    WORD@532..537 "pages"
    WHITESPACE@537..545 "        "
    EQ@545..546 "="
    WHITESPACE@546..547 " "
    L_CURLY@547..548 "{"
    WORD@548..557 "3027-3036"
    R_CURLY@557..558 "}"
    COMMA@558..559 ","
    WHITESPACE@559..564 "\n    "
    WORD@564..574 "indextitle"
    WHITESPACE@574..577 "   "
    EQ@577..578 "="
    WHITESPACE@578..579 " "
    L_CURLY@579..580 "{"
    WORD@580..586 "Effect"
    WHITESPACE@586..587 " "
    WORD@587..589 "of"
    WHITESPACE@589..590 " "
    WORD@590..604 "immobilization"
    WHITESPACE@604..605 " "
    WORD@605..607 "on"
    WHITESPACE@607..608 " "
    WORD@608..617 "catalytic"
    WHITESPACE@617..618 " "
    WORD@618..633 "characteristics"
    R_CURLY@633..634 "}"
    COMMA@634..635 ","
    WHITESPACE@635..636 "\n"
    R_CURLY@636..637 "}"

